BranchMetrics/ios-branch-deep-linking-attribution
# Sdk 1934 add public method for branch api base url

## Reference
SDK-1934 -- Expose a public method to change the Branch API base url.

## Summary
Added following APIs in BranchPluginSupport class for setting up Server and CDN Urls-
```
      + (void)setAPIUrl:(NSString *)url;
     + (void)setCDNBaseUrl:(NSString *)url;
```
**Note:** If url passed is not valid or it doesn't start with "http" or "https" then default Urls defined BNCConfig are used.

## Type Of Change


## Testing Instructions
==>  Run Unit tests. Following Tests are added for above APIs -
```
- testSetAPIURL_Example
- testSetAPIURL_InvalidHttp 
- testSetAPIURL_InvalidEmpty 
- testSetCDNBaseURL_Example 
- testSetCDNBaseURL_InvalidHttp 
- testSetCDNBaseURL_InvalidEmpty

```
**Note**:  Tests for Null url are not added. Because of non_null attribute assigned to class BranchPluginSupport, API was giving warning on passing Null param. So I removed tests for null url.

==> In any test app, set URL https://dashboard.stage.branch.io and then check in Server requests logs, if this new url is used.
`[BranchPluginSupport setAPIUrl:@"https://dashboard.stage.branch.io"];`

cc @BranchMetrics/saas-sdk-devs for visibility.
